src/components/Tables/Filters.tsx

```diff
@@ -1,36 +1,35 @@
 import React from "react";
 import { TableHeaders } from "@/components/Tables/Table";
+import { Database } from "@/databaseTypesFile";
+import { PostgrestFilterBuilder } from "@supabase/postgrest-js";
+
+export enum PaginationType {
+    Server = "SERVER",
+    Client = "CLIENT",
+}
+
+export type MethodConfig<Data, State> =
+    | {
+          method: (
+              query: PostgrestFilterBuilder<Database["public"], any, any>,
+              state: State
+          ) => PostgrestFilterBuilder<Database["public"], any, any>;
+          methodType: PaginationType.Server;
+      }
+    | {
+          method: (row: Data, state: State, key: keyof Data) => boolean;
+          methodType: PaginationType.Client;
+      };
 
 export interface Filter<Data, State> {
-    shouldFilter: (data: Data, state: State) => boolean;
+    key: keyof Data;
     filterComponent: (state: State, setState: (state: State) => void) => React.ReactElement;
     state: State;
     initialState: State;
+    methodConfig: MethodConfig<Data, State>;
+    areStatesIdentical: (stateA: State, stateB: State) => boolean;
 }
 
-export interface KeyedFilter<Data, Key extends keyof Data, State> extends Filter<Data, State> {
```

Comment:
we never really needed the distinction between a Filter and a KeyedFilter, we should just assign a key to every filter, because filters are components that end up in an array, so we need to give them keys.